consensus/src/quorum_store/utils.rs

```diff
@@ -389,26 +517,127 @@ impl ProofQueue {
+        counters::NUM_TOTAL_TXNS_LEFT_ON_UPDATE.observe(self.remaining_txns_with_duplicates as f64);
         counters::NUM_TOTAL_PROOFS_LEFT_ON_UPDATE.observe(self.remaining_proofs as f64);
         counters::NUM_LOCAL_TXNS_LEFT_ON_UPDATE.observe(self.remaining_local_txns as f64);
         counters::NUM_LOCAL_PROOFS_LEFT_ON_UPDATE.observe(self.remaining_local_proofs as f64);
+        let remaining_txns_without_duplicates = self.remaining_txns_without_duplicates();
+        counters::NUM_UNIQUE_TOTAL_TXNS_LEFT_ON_UPDATE
+            .observe(remaining_txns_without_duplicates as f64);
+        //count the number of transactions with more than one batches
+        counters::TXNS_WITH_DUPLICATE_BATCHES.set(
+            self.txn_summary_to_batches
+                .iter()
+                .filter(|(_, batches)| batches.len() > 1)
+                .count() as i64,
+        );
 
-        (self.remaining_txns, self.remaining_proofs)
+        counters::TXNS_IN_PROOF_QUEUE.set(self.txn_summary_to_batches.len() as i64);
+
+        // count the number of batches with proofs but without txn summaries
+        counters::PROOFS_WITHOUT_BATCH_DATA.set(
+            self.batch_to_proof
+                .iter()
+                .map(|(batch_key, proof)| {
+                    if proof.is_some() && !self.batches_with_txn_summary.contains(batch_key) {
+                        1
+                    } else {
+                        0
+                    }
+                })
+                .sum::<i64>(),
+        );
+
+        counters::PROOFS_IN_PROOF_QUEUE.set(
+            self.batch_to_proof
+                .values()
+                .map(|proof| if proof.is_some() { 1 } else { 0 })
+                .sum::<i64>(),
+        );
+        (remaining_txns_without_duplicates, self.remaining_proofs)
     }
 
     // Mark in the hashmap committed PoS, but keep them until they expire
-    pub(crate) fn mark_committed(&mut self, batches: Vec<BatchInfo>) {
-        for batch in batches {
-            let batch_key = BatchKey::from_info(&batch);
+    fn mark_committed(&mut self, batches: Vec<BatchInfo>) {
+        for batch in &batches {
+            let batch_key = BatchKey::from_info(batch);
             if let Some(Some((proof, insertion_time))) = self.batch_to_proof.get(&batch_key) {
                 counters::pos_to_commit(
                     proof.gas_bucket_start(),
                     insertion_time.elapsed().as_secs_f64(),
                 );
                 self.dec_remaining(&batch.author(), batch.num_txns());
             }
-            self.batch_to_proof.insert(batch_key, None);
+            self.batch_to_proof.insert(batch_key.clone(), None);
+            self.batches_with_txn_summary.remove(&batch_key);
+            self.txn_summary_to_batches.retain(|_, batches| {
+                batches.remove(&batch_key);
+                !batches.is_empty()
+            });
+        }
+    }
+
+    pub async fn start(
```

Comment:
Do we really need a separate loop for ProofQueue, outside of ProofManager? Are there any issues without this loop?

Reply:
Both the batch coordinator and proof manager need to interact with proof queue. The batch coordinator sends the created/received batches to the proof queue. So, semantically I thought proof queue should be its own thread. 

Proof queue could also be merged into proof manager. In that case, the batch coordinator will send the created/received batches to the proof manager. Do you prefer this way instead?